index.d.ts

```diff
@@ -23,9 +23,20 @@ import { Readable } from 'stream'
 type Mercurius = typeof mercurius
 
 declare namespace mercurius {
-  export interface PubSub {
-    subscribe<TResult = any>(topics: string | string[]): Promise<Readable & AsyncIterableIterator<TResult>>;
-    publish<TResult = any>(event: { topic: string; payload: TResult }, callback?: () => void): void;
+  export type PubSubPublishArgsByKey = {
+    [key: string]: any;
+  };
```

Comment:
This should probably mimic eg [`MercuriusContext`](https://github.com/mercurius-js/mercurius/blob/a45fd77b7529cd763fba367b2f21eb0c583e5fd6/index.d.ts#L31-L41) and be one can extend, rather than (or in addition) to being needed to specify like `PubSub<MyTopics>` – that way it will work automatically in all the places that use the `PubSub` type.

The name should also be in the style of the other interfaces.

```suggestion
  export interface PubSubTopics {
    [topic: string]: any;
  }
```

Reply:
Fixed.